game-core/src/main/java/games/strategy/engine/framework/ServerGame.java

```diff
@@ -438,22 +439,24 @@ private void runStep(final boolean stepIsRestoredFromSavedGame) {
     if (gameData.getSequence().next()) {
       gameData.getHistory().getHistoryWriter().startNextRound(gameData.getSequence().getRound());
       saveGame(gameData.getSequence().getRound() % 2 == 0
-          ? SaveGameFileChooser.getEvenRoundAutoSaveFile(HeadlessGameServer.headless())
-          : SaveGameFileChooser.getOddRoundAutoSaveFile(HeadlessGameServer.headless()));
+          ? SaveGameFileChooser.getEvenRoundAutoSaveFile(headless)
+          : SaveGameFileChooser.getOddRoundAutoSaveFile(headless));
     }
     if (autoSaveThisDelegate && !currentStep.getName().endsWith("Move")) {
-      autoSaveAfter(currentDelegate);
+      autoSaveAfter(currentDelegate, headless);
     }
   }
 
-  private void autoSaveAfter(final GameStep gameStep) {
-    saveGame(SaveGameFileChooser.getAfterStepAutoSaveFile(gameStep.getName()));
+  private void autoSaveAfter(final String stepName, final boolean headless) {
+    saveGame(SaveGameFileChooser.getAfterStepAutoSaveFile(
+        headless ? (stepName.endsWith("NonCombatMove") ? "NonCombatMove" : "CombatMove") : stepName, headless));
   }
 
-  private void autoSaveAfter(final IDelegate delegate) {
+  private void autoSaveAfter(final IDelegate delegate, final boolean headless) {
     final String typeName = delegate.getClass().getTypeName();
     final String stepName = typeName.substring(typeName.lastIndexOf('.') + 1).replaceFirst("Delegate$", "");
-    saveGame(SaveGameFileChooser.getAfterStepAutoSaveFile(stepName));
+    // can only get here for a headed process
```

Comment:
If this statement is true, should we remove the second parameter from this method and always call `getAfterStepAutoSaveFile()` with `headless` equal to `false`?